llvm/lib/Analysis/InlineOrder.cpp

```diff
@@ -273,7 +364,14 @@ llvm::getInlineOrder(FunctionAnalysisManager &FAM, const InlineParams &Params) {
   case InlinePriorityMode::CostBenefit:
     LLVM_DEBUG(
         dbgs() << "    Current used priority: cost-benefit priority ---- \n");
-    return std::make_unique<PriorityInlineOrder<CostBenefitPriority>>(FAM, Params);
+    return std::make_unique<PriorityInlineOrder<CostBenefitPriority>>(FAM,
+                                                                      Params);
+
+  case InlinePriorityMode::TopDown:
+    LLVM_DEBUG(
+        dbgs() << "    Current used priority: top-down priority ---- \n");
+    return std::make_unique<PriorityInlineOrder<TopDownPriority>>(FAM, Params,
```

Comment:
With a new subclass you would return here:
`return std::make_unique<TopDownInlineOrder(FAM, Params, M);`